### PR TITLE
[doc] Porting notes to 2.0

### PR DESCRIPTION
**Pull request by gabrielebndn**
Hi,
I think it's time to make some porting notes.
Following a talk with @nmansard, I included them in the doc.

I just put the first few things that came to my mind. If anybody can think of anything else to add (especially @jcarpent), they can add here.

I preferred to make this PR to the devel branch, rather than topic/doc-v2, because this way it will be immediately available.

### CODE REVIEW

File: doc/_porting.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+The vast majority of the changes took place in C++.
+
+# Changes in C++
+However, it should not change a lot from the user's point of view.
+
+## Namespace
+The most important change is the namespace.
+Now, the top-level pinocchio namespace is not `se3` anymore, but `pinocchio`.
+
+Therefore, all occurrence of `se3` in your code should be switched to `pinocchio`.
```
Comment (jcarpent) [inline doc/_porting.md]: 'should be replaced by`

File: doc/_porting.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+If you feel like ''pinocchio'' is too long to type but you do not want to do `using namespace pinocchio`,
+you can use
+```
+namespace pin = pinocchio;
+```
+
+Your code will not compile you try to use namespace `se3`.
```
Comment (jcarpent) [inline doc/_porting.md]: missing if

File: doc/_porting.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+In order to mak it work, you need to compile it with the following flag
```
Comment (jcarpent) [inline doc/_porting.md]: to make

File: doc/_porting.md [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+```
+-DPINOCCHIO_ENABLE_COMPATIBILITY_WITH_VERSION_1
+```
+
+## Deprecated macros
+
+The following marcos are not emplyed anymore
```
Comment (jcarpent) [inline doc/_porting.md]: employed anymore:

**Comment by jcarpent** [comment]
Ok.
I think it can be worth to use **Pinocchio** for naming the library and classic pinocchio for naming the namespace.
Can you apply this rule?

**Comment by gabrielebndn** [comment]
Done

**Pull request merged**